nautobot_netbox_importer/generator/base.py

```diff
@@ -105,71 +98,47 @@ class InternalFieldType(Enum):
     InternalFieldType.UUID_FIELD: UUID,
 }
 
-INTEGER_AUTO_FIELD_TYPES: Iterable[InternalFieldType] = (
-    InternalFieldType.AUTO_FIELD,
-    InternalFieldType.BIG_AUTO_FIELD,
+# Fields to auto add to source and target wrappers
+AUTO_ADD_FIELDS = (
+    "content_types",
+    "status",
 )
 
-INTEGER_INTERNAL_TYPES: Iterable[InternalFieldType] = (
-    InternalFieldType.AUTO_FIELD,
-    InternalFieldType.BIG_AUTO_FIELD,
-    InternalFieldType.BIG_INTEGER_FIELD,
-    InternalFieldType.INTEGER_FIELD,
-    InternalFieldType.POSITIVE_INTEGER_FIELD,
-    InternalFieldType.POSITIVE_SMALL_INTEGER_FIELD,
-    InternalFieldType.SMALL_INTEGER_FIELD,
+EMPTY_VALUES = (
+    "",
+    NOT_PROVIDED,
+    None,
+    [],
+    dict,  # CustomFieldData field has `dict` class as the default value
+    set(),
+    tuple(),
+    {},
 )
 
-REFERENCE_INTERNAL_TYPES: Iterable[InternalFieldType] = (
-    InternalFieldType.FOREIGN_KEY,
-    InternalFieldType.FOREIGN_KEY_WITH_AUTO_RELATED_NAME,
-    InternalFieldType.GENERIC_FOREIGN_KEY,
-    InternalFieldType.MANY_TO_MANY_FIELD,
-    InternalFieldType.ONE_TO_ONE_FIELD,
-    InternalFieldType.ROLE_FIELD,
-    InternalFieldType.STATUS_FIELD,
-    InternalFieldType.TREE_NODE_FOREIGN_KEY,
-)
-
-DONT_IMPORT_TYPES: Iterable[InternalFieldType] = (
-    InternalFieldType.DO_NOT_IMPORT_LAST_UPDATED,
-    InternalFieldType.NOT_FOUND,
-    InternalFieldType.PRIVATE_PROPERTY,
-    InternalFieldType.READ_ONLY_PROPERTY,
-)
-
-EMPTY_VALUES = [None, set(), tuple(), {}, [], "", NOT_PROVIDED]
 
-
-# pylint: disable=too-many-return-statements
 def get_nautobot_field_and_type(
     model: NautobotBaseModelType,
     field_name: str,
-) -> Tuple[Optional[NautobotField], InternalFieldType]:
+) -> Tuple[Optional[DjangoField], InternalFieldType]:
```

Comment:
why do we move from NautoboField to DjangoField?

Reply:
Related to the refactoring of `GenericForeignKey` fields. These fields are not needed anymore and it was the only fields that were not standard DjangoField. So it was possible to remove this type and replace it with the `DjangoField` type only.

https://github.com/nautobot/nautobot-app-netbox-importer/pull/123/files#diff-6b479358489e04d98598db52da0a5cce8c2636d21202d851e16589d38820e84eL42